scripts/build-server

```diff
@@ -8,6 +8,14 @@ cd $(dirname $0)/..
 mkdir -p bin
 [ "$(uname)" != "Darwin" ] && LINKFLAGS="-extldflags -static -s"
 
-RKE_VERSION=$(grep -m1 '^github.com/rancher/rke' vendor.conf | awk '{print $2}')
+if echo "$VERSION" | grep -q -e '^v.*' ; then 
+    UI_INDEX="local"
+fi
 
-CGO_ENABLED=0 go build -i -tags k8s -ldflags "-X main.VERSION=$VERSION -X github.com/rancher/rancher/pkg/settings.RKEVersion=$RKE_VERSION $LINKFLAGS" -o bin/rancher
+# Inject Setting values
+DEFAULT_VERSIONS=`\
+  INJECT_UI_INDEX=${UI_INDEX:-"https://releases.rancher.com/ui/latest2/index.html"} \
+  INJECT_RKE_VERSION="$(grep -m1 '^github.com/rancher/rke' vendor.conf | awk '{print $2}')" \
+  go run ./pkg/settings/generator/main.go`
```

Comment:
now that i fully understand this, i have to ask, why not just do:
```
DEFAULT_VERSIONS="{\"uiIndex\": \"${UI_INDEX}\", \"rkeVersion\": \"${RKE_VERSION}\"}"
```

I feel like the go program adds a lot of extra complexity. What is it buying us? I realize the above is not terribly readable, but I think it is easier to grok and maintain than the go program

Reply:
(my variable substitutions are psuedo-code, youd have to do the right thing for those of course)

Reply:
I want to make `DEFAULT_VERSIONS` more easy to add and more easy to read. The variable is less complex than a JSON string in shell script and also easier to maintain. That is why I want to use environment variable instead of JSON.